connectors/src/connectors/webcrawler/temporal/activities.ts

```diff
@@ -191,26 +191,22 @@ export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
 
   await crawler.addRequests([webCrawlerConfig.url]);
 
-  const crawlPromise = crawler.run([webCrawlerConfig.url]);
-  let isResolved = false;
-  crawlPromise
-    .then(() => {
-      isResolved = true;
+  let crawlResolved = false;
+  const crawlPromise = crawler
+    .run([webCrawlerConfig.url])
+    .then((result) => {
+      crawlResolved = true;
+      return result;
     })
     .catch((e) => {
-      logger.error(
-        {
-          error: e,
-          connectorId: connector.id,
-          configId: webCrawlerConfig.id,
-        },
-        "Webcrawler error while crawling"
-      );
+      crawlResolved = true;
+      throw e;
     });
   try {
-    while (!isResolved) {
+    while (!crawlResolved) {
       await Context.current().sleep(1000); // allows the activity to react to cancellation
     }
+    await crawlPromise; // will throw if the crawl failed
```

Comment:
Not super easy to follow :-/

And not completely convinced it will even work to interrupt the activity? What would interrupt the promise code execution here?

Reply:
As the code start executing it’s handled by node right? And there is surely no process.exit involved here?

Reply:
the promise is already resolved at this line, so there's no activity interruption topic

Above, Context.current().sleep is a call to temporal, that's where temporal can throw a CancelFailure that will be repercuted here => aborting the crawl (via the crawler's method), and throwing (which in turn will throw an activity error, which will warn us

Agreed it's not that easy to follow, but not sure how to make it clearer apart from better commenting

Reply:
But it won’t stop the crawler will it?